### PR TITLE
Issue 4364 - Query Store Options for 2017

### PR DESCRIPTION
**Pull request by tboggiano**
Adding the max plan per queries and wait stats capture mode.  As noted code is in here for 2019 features as well but commented out, I'll come back and fix that when it is RTM.  

## Type of Change

 - [ ] Bug fix (non-breaking change, fixes #<enter issue number>)
 - [ X ] New feature (non-breaking change, adds functionality)
 - [ ] Breaking change (effects multiple commands or functionality)
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 

### Purpose
 
To add the new parameters from 2017 max plans per query and wait stats capture mode.
### Approach

Allows you to see, copy, and set those options.
### Commands to test

copy and get functions are the same.  Set-DbaDbQueryStoreOptions -SQLInstance . -Database DBA -MaxPlansPerQuery 100 -WaitStatsCaptureMode Off

### CODE REVIEW

**Comment by potatoqualitee** [comment]
thanks so much! looks like one test needs to be updated with the added params. would you like to do that or would you like me to?

File: functions/Copy-DbaDbQueryStoreOption.ps1 [hunk 1]
```diff
@@ -147,19 +158,58 @@ function Copy-DbaDbQueryStoreOption {
...
+                                    CollectionInterval   = $SourceQSConfig.StatisticsCollectionIntervalInMinutes;
+                                    MaxSize              = $SourceQSConfig.MaxStorageSizeInMB;
+                                    CaptureMode          = $SourceQSConfig.QueryCaptureMode;
+                                    CleanupMode          = $SourceQSConfig.SizeBasedCleanupMode;
+                                    StaleQueryThreshold  = $SourceQSConfig.StaleQueryThresholdInDays;
+                                    MaxPlansPerQuery     = $QueryStoreOptions.MaxPlansPerQuery;
+                                    WaitStatsCaptureMode = $QueryStoreOptions.WaitStatsCaptureMode;
+                                }
                             }
+                            #For 2019 RTM
```
Comment (potatoqualitee) [inline functions/Copy-DbaDbQueryStoreOption.ps1]: I know it's just comments but I have OCD ;) Please switch to multi-line comments, which are available in PowerShell as `<#`  and `#>`

Comment (tboggiano) [reply functions/Copy-DbaDbQueryStoreOption.ps1]: I like OCD :) I can update the test.  Gives me practice.

Comment (potatoqualitee) [reply functions/Copy-DbaDbQueryStoreOption.ps1]: Thanks much, looks good. that latest failure is appveyor being a 🍑 , and not due to your changes. I'll be merging shortly!

**Comment by potatoqualitee** [comment]
hey tracy - my apologies for the delay. i'm knee deep in another project right now and a bit distracted.

After confirming with you that the code would work with current 2019 bits, I've made some modifications as can be seen in this commit https://github.com/sqlcollaborative/dbatools/pull/5926/commits/4928458eb885af5313c9ce86f1caa1541181dcf0

I also did a lil bit of cleanup of the command in general while I was in it (removed unneeded semicolons, etc). I did find two typos - if you can confirm that my changes were what was intended, please let me know. Also, please test the commands and let me know as well. 

If you paste the commands you use to test here, I'll add them to our integration tests in Pester and ping you with that commit to show you what I did. you are also welcomed to add the integration tests, I'm just not available to help much at this moment.

Chrissy

**Comment by potatoqualitee** [comment]
k will merge then update 💯

**Pull request merged**